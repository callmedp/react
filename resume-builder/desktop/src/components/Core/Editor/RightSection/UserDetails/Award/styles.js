import { StyleSheet } from 'aphrodite';

export const styles = StyleSheet.create({
     h2 :{
    margin: [0,10, !important],
    fontWeight: 1.8rem,
    fontWeight: 700
  },

  iconAwardsCursor: {
    cursor: pointer
  },

addButtonRight: {
    marginLeft: auto,
    marginRight: 20
        
    &:before {
      content: '+';
      font-size: 1.8rem;
      padding-right: 5px;
    }
  },

  tillToday: {
    color: $txt2;
    margin: 10px 0 0 0;
    display: block;
    text-align: right;

    input: {
      margin-right: 5px;
    }
  }
})