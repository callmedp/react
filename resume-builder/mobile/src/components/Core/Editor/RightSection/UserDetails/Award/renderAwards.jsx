import React from 'react';
import {datepicker, renderField, renderTextArea} from "../../../../../FormHandler/formFieldRenderer.jsx";
import {Field} from "redux-form";
import {awardNewData} from "../../../../../../Utils/addnewData"

export const renderAwards = ({
                                fields, 
                                meta: {touched, error, submitFailed},
                                handleSubmit,
                                handleAddition,
                                deleteAward,
                                changeOrderingUp,
                                changeOrderingDown,
                                editHeading,
                                heading,
                                updateInputValue,
                                editHeadingClick,
                                context,
                            }) => {
    return (
        
        <div className="buildResume__wrap">
            <div className="buildResume__heading heading">
                <div className="heading__info">
                    {!editHeading ?
                        <React.Fragment>
                            <h1>{heading}</h1>
                            <i className="sprite icon--edit" onClick={editHeadingClick.bind(true)}></i>
                        </React.Fragment>:
                        <React.Fragment>
                            <input type="text" autoFocus defaultValue={heading} onBlur={(e)=>updateInputValue('blur',e)}
                                onKeyDown={(e)=>updateInputValue('keyPress',e)} maxLength="20"/>
                            <i className="sprite icon--editTick"></i>
                        </React.Fragment>
                    }
                </div>
                {!editHeading ?
                    <button role="button" onClick={handleSubmit(handleAddition.bind(this,fields,awardNewData(fields),100,'award'))}
                        type={'button'} className="btn btn__round btn--outline">+ Add new</button>:''
                }
            </div>

            {fields.map((member, index) => {
            return(
                <div className="form-wrap" key={index} id={`award${index}`}>
                    <div className="subHeading pb-0">
                        <h2>{fields.get(index).title || 'Award'}</h2>
                        <ul className="subHeading__control">
                            <li className="subHeading__delete">
                                <span className={"sprite icon--delete " +(fields.length === 1 && !fields.get(index).id ? "hide":"")}  role="button"
                                    onClick={(event) => deleteAward(index, fields, event)}></span>
                            </li>
                            {index === 0 ? '':
                                <li className="subHeading__btn"
                                    onClick={(event) =>{fields=changeOrderingUp(index, fields, event);context.setState({fields})}}>
                                    <i className="sprite icon--upArrow"></i>
                                </li>
                            }
                            {index === fields.length-1 ? '':
                                <li className="subHeading__btn"
                                    onClick={(event) =>{fields=changeOrderingDown(index, fields, event);context.setState({fields})}}>
                                    <i className="sprite icon--downArrow"></i>
                                </li>
                            }
                        </ul>
                    </div>

                    <ul className="form pb-0">

                        <li className="form__group">
                            <Field component={renderField} label={"Title"}  type={"text"} name={`${member}.title`} prepend={true}
                                id={`${member}.title`} iconClass={"sprite icon--education-grey"} className="form__input"/>
                        </li>

                        <li className="form__group">
                            <Field component={datepicker} label={"Date"}  type={"date"} 
                                name={`${member}.date`} id={`${member}.date`}/>
                        </li>

                        <li className="form__group">
                            <label className="form__label" htmlFor="summary">Summary</label>
                            <div className="input-group">
                                <div className="input-group__prepend">
                                    <span className="input-group__text">
                                        <i className="sprite icon--course-type"></i>
                                    </span>
                                </div>
                                <Field component={renderTextArea} type={"textarea"} className={'form__input'}
                                    className="form__input" name={`${member}.summary`}/>
                            </div>
                        </li>
                    </ul>
                </div>
            )})}
        </div>
            
    )
}

export default renderAwards