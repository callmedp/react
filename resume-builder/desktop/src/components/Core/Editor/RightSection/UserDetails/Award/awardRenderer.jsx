import {
    Accordion,
    AccordionItem,
    AccordionItemButton,
    AccordionItemHeading,
    AccordionItemPanel
} from "react-accessible-accordion";
import styles from './award.scss'
import {Field} from "redux-form";
import {renderField, renderTextArea, renderSelect} from "../../../../../FormHandler/formFieldRenderer";
import React from "react";
import {yearList} from "../../../../../../Utils/yearList";

export const AwardRenderer = ({
                                  fields,
                                  loader,
                                  meta: {touched, error, submitFailed},
                                  handleSubmit,
                                  deleteAward,
                                  handleAddition,
                                  handleAccordionClick,
                                  changeOrderingUp,
                                  changeOrderingDown,
                                  openedAccordion,
                                  isEditable,
                                  handleInputValue,
                                  editHeading,
                                  saveTitle,
                                  entityName,
                                  expanded
                              }) => {
    return (
        <div>
            {/*{!!(loader) &&*/}
            {/*<LoaderSection/>*/}
            {/*}*/}
            <section className={'head-section'}>
                <span className={'icon-box'}><i className={'icon-awards1'}/></span>
                {!!(!isEditable) ?
                    <h2>{entityName}
                    </h2> :
                    <React.Fragment>
                        <input autoFocus type="text" name="" defaultValue={entityName}
                               onChange={(event) => handleInputValue(event.target.value || entityName)} maxLength="20"/>
                        <span onClick={(event) => saveTitle(event)} className="icon-tick"/>
                    </React.Fragment>
                }
                <span onClick={() => editHeading()}
                      className={!!(!isEditable) ? "icon-edit " + styles['icon-edit__cursor'] : ''}/>
                <button onClick={handleSubmit((values) => {
                    handleAddition(fields, error)
                })}
                        type={'button'}
                        className={'add-button ' + styles['add-button__right']}>Add new
                </button>

            </section>
            <section className="right-sidebar-scroll" id="award">
                <ul>
                    <Accordion
                        onChange={(value) => handleAccordionClick(value)}
                        allowZeroExpanded={false}
                        allowMultipleExpanded={true}
                        preExpanded={[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]}
                    >
                        {
                            fields.map((member, index) => {
                                return (
                                    <li key={index}>
                                        <section className="info-section">
                                            <AccordionItem uuid={index} id={`award${index}`}>
                                                <AccordionItemHeading>
                                                    <AccordionItemButton>
                                                        <div className="flex-container">
                                                            <h3 className={"add-section-heading"}>{fields.get(index).title || 'Award'}</h3>
                                                            <span
                                                                className={expanded.indexOf(index) > -1 ? "opened-accordion" : "closed-accordion"}></span>
                                                            <div className="addon-buttons mr-10">
                                                                <span
                                                                    onClick={(event) => deleteAward(index, fields, event)}
                                                                    className="icon-delete mr-15"/>
                                                                {index !== 0 &&
                                                                <span
                                                                    onClick={(event) => changeOrderingUp(index, fields, event)}
                                                                    className="icon-ascend mr-5"/>
                                                                }
                                                                {
                                                                    index !== fields.length - 1 &&
                                                                    < span
                                                                        onClick={(event) => changeOrderingDown(index, fields, event)}
                                                                        className="icon-descend"/>
                                                                }
                                                            </div>
                                                        </div>
                                                    </AccordionItemButton>
                                                </AccordionItemHeading>
                                                <AccordionItemPanel>
                                                    <div className="flex-container">
                                                        <fieldset className="error">
                                                            <label>Title</label>
                                                            <Field
                                                                autoFocus={true}
                                                                maxLength={'50'}
                                                                iconClass={'icon-awards-gr'}
                                                                component={renderField}
                                                                type={"text"}
                                                                name={`${member}.title`}
                                                                className="input-control"/>
                                                        </fieldset>
                                                        <fieldset>
                                                            <label>Year</label>
                                                            <Field
                                                                iconClass={'icon-date'}
                                                                component={renderSelect}
                                                                maxDateAllowed={true}
                                                                options={yearList}
                                                                type={"date"}
                                                                className={'input-control'}
                                                                name={`${member}.date`}/>
                                                        </fieldset>
                                                    </div>
                                                    <div className="flex-container">
                                                        <fieldset>
                                                            <label>Summary</label>
                                                            <Field
                                                                noIcon={true}
                                                                maxLength={'300'}
                                                                component={renderTextArea} type={"textarea"}
                                                                name={`${member}.summary`}
                                                                className="input-control"/>
                                                        </fieldset>
                                                    </div>
                                                </AccordionItemPanel>
                                            </AccordionItem>
                                        </section>
                                    </li>
                                )
                            })
                        }
                    </Accordion>
                </ul>
            </section>


        </div>
    )
}
