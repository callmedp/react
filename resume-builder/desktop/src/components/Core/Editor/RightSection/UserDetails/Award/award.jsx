import React, {Component} from 'react';
import {FieldArray, reduxForm} from "redux-form";
import * as actions from "../../../../../../store/award/actions";
import {connect} from "react-redux";
import moment from "moment";
import validate from "../../../../../FormHandler/validations/award/validate";
import {AwardRenderer} from "./awardRenderer";
import {scrollOnErrors} from "../../../../../../Utils/srollOnError"
import {scroller} from "react-scroll/modules";
import SavePreviewButtons from '../../../../../Common/SavePreviewButtons/savePreviewButtons';

class Award extends Component {
    constructor(props) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleAccordionClick = this.handleAccordionClick.bind(this);
        this.handleAddition = this.handleAddition.bind(this);
        this.deleteAward = this.deleteAward.bind(this);
        this.updateInfoBeforeLoss = this.updateInfoBeforeLoss.bind(this);
        this.state = {
            active: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
            submit: false

        }
        this.props.currentForm('award');
    }

    componentDidMount() {
        this.props.fetchUserAward()
    }

    async componentDidUpdate(prevProps){
        const {ui:{previewClicked},previewButtonClicked,history} = this.props;
        if(previewClicked !== prevProps.ui.previewClicked && previewClicked){
            await this.updateInfoBeforeLoss()
            this.setState({submit:true})
            previewButtonClicked(false)
            history.push('/resume-builder/preview/')
        }
    }

    async updateInfoBeforeLoss(){
        let { initialValues, formData: {award: {values, syncErrors}}} = this.props;
        let error = false;
        ((syncErrors && syncErrors['list']) || []).map(el => Object.keys(el || {}).map(key => (!!el[key] ? error = true : false)));
        if (!error && !this.state.submit && JSON.stringify(initialValues)!==JSON.stringify(values)) await this.props.bulkUpdateOrCreate(values && values['list'])
    }


    componentWillUnmount() {
        this.updateInfoBeforeLoss()
    }

    async handleSubmit(values, entityLink) {
         const {generateResumeAlert,bulkUpdateOrCreate,history} = this.props
        const {list} = values;
        if (list.length) {
            // skip the api call if there is a certain field which is required but empty (We skipped validation intentionally)
            let skipApiCall = false;
            
            list.map(el => {
                if(! el.title){
                    skipApiCall = true;
                }
                return; 
            })

            if(!skipApiCall){
                await bulkUpdateOrCreate(list);
            }
        }

        this.setState({
            submit: true
        })

        if (entityLink) history.push(entityLink);
            else{
                generateResumeAlert()
            }
    }

    handleAddition(fields, error) {
        const listLength = fields.length;
        this.props.sendTrackingInfo('right_section_add',1)
        fields.push({
            "candidate_id": '',
            "id": '',
            "title": '',
            "date": '',
            "summary": '',
            order: listLength
        })

        scroller.scrollTo(`award${fields.length - 1}`, {
            duration: 800,
            delay: 0,
            smooth: 'easeInOutQuad',
            offset: 200,
            containerId: 'award'
        })
        this.props.eventClicked({
            'action':'AddNew',
            'label':'Awards'
        })
    }

    deleteAward(index, fields, event) {
        this.props.sendTrackingInfo('right_section_delete',1)
        event.stopPropagation();
        const award = fields.get(index);
        fields.remove(index);
        if (award && award.id) {
            this.props.removeAward(award.id)
        }
    }

    handleAccordionClick(value, fields) {
        this.setState({active: value})
    }

    render() {
        const {
            handleSubmit,userInfo:{order_data}, ui: {loader}, saveTitle, editHeading,eventClicked,
            isEditable, entityName, handleInputValue, nextEntity, showAlertModal,history, changeOrderingDown, changeOrderingUp, showAlertMessage
        } = this.props;

        return (
            <form onSubmit={handleSubmit((values) => this.handleSubmit(values, nextEntity))}>
                <FieldArray name="list"
                            loader={loader}
                            handleSubmit={handleSubmit}
                            handleAccordionClick={this.handleAccordionClick}
                            handleAddition={this.handleAddition}
                            deleteAward={this.deleteAward}
                            changeOrderingUp={changeOrderingUp}
                            changeOrderingDown={changeOrderingDown}
                            component={AwardRenderer}
                            saveTitle={(event) => saveTitle(event, 7)}
                            editHeading={() => editHeading(7)}
                            entityName={entityName}
                            isEditable={isEditable}
                            handleInputValue={handleInputValue}
                            expanded={this.state.active}
                            showAlertMessage={showAlertMessage}
                />
                <SavePreviewButtons 
                        showAlertModal={showAlertModal} context={this} history={history} order_data={order_data} form_name={'Awards'}
                        nextEntity={nextEntity} updateInfoBeforeLoss={this.updateInfoBeforeLoss} eventClicked={eventClicked}
                    />

            </form>
        )
    }
}


export const AwardForm = reduxForm({
    form: 'award',
    validate,
    enableReinitialize: true,
    onSubmitFail: (errors) => scrollOnErrors(errors, 'award', -100),
})(Award);


const mapStateToProps = (state) => {
    return {
        initialValues: state.award,
        ui: state.ui
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        "onSubmit": (userAward) => {
            const {date} = userAward;

            userAward = {
                ...userAward,
                ...{
                    date: (date && moment(date).format('YYYY-MM-DD')) || '',
                }
            };
            return new Promise((resolve, reject) => {
                return dispatch(actions.updateUserAward({userAward, resolve, reject}));
            })
        },
        "fetchUserAward": () => {
            return dispatch(actions.fetchUserAward())
        },
        "removeAward": (awardId) => {

            return dispatch(actions.deleteAward(awardId))
        },

        "bulkUpdateOrCreate": (listItems) => {
            listItems = (listItems || []).map((userAward, index) => {
                const {date} = userAward;
                if (!userAward['id']) delete userAward['id'];
                userAward = {
                    ...userAward,
                    ...{
                        date: (date && date.value && Number(date.value)) || null,
                        order: index
                    }
                };
                return userAward;
            });
            return new Promise((resolve, reject) => {
                return dispatch(actions.bulkUpdateOrCreateUserAward({list: listItems, resolve, reject}))
            })

        },
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(AwardForm);
