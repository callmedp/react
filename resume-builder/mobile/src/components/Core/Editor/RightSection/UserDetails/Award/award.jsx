import React, {Component} from 'react';
import {reduxForm, FieldArray} from "redux-form";
import * as actions from "../../.../../../../../../store/award/actions";
import {connect} from "react-redux";
import validate from "../../../../../FormHandler/validtaions/award/validate"
import moment from "moment";
import PreviewModal from "../../../Preview/changeTemplateModal";
import renderAwards from "./renderAwards"
import {siteDomain} from "../../../../../../Utils/domains";
import {scrollOnErrors} from "../../../../../../Utils/srollOnError"
import BottomCTC from '../../../../../Common/BottomCTC/bottom-ctc';


class Award extends Component {

    constructor(props){
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.deleteAward = this.deleteAward.bind(this);
        this.state = {
            'editHeading': false,
            'heading' : '',
            'submit'  : false
        }
        this.editHeadingClick = this.editHeadingClick.bind(this);
        this.updateInfoBeforeLoss = this.updateInfoBeforeLoss.bind(this)
    }

    componentDidMount() {
        this.props.fetchUserAward()
        if (this.props.personalInfo.entity_preference_data.length) {
            this.setState({heading : this.props.personalInfo.entity_preference_data[6].entity_text})
        }
    }

    async updateInfoBeforeLoss(){
        if(!this.state.submit){
            const {initialValues} =this.props
            const form_data = this.props.info.form.award;
            let error = false
            let error_values =form_data["syncErrors"]
            if(error_values){
                for(let i of  error_values['list']){
                    for(let j of Object.keys(i)){
                        if(i[j]){
                            error =true
                            break;
                        }
                    }
                }
            }
            if(!error && JSON.stringify(initialValues)!==JSON.stringify(form_data['values'])){
                
                const values = this.props.handleOrdering(form_data['values'])
                await this.props.bulkUpdateUserAward(values.list)
            }
        }  
    }

    componentWillUnmount() {
        this.updateInfoBeforeLoss()
    }
    

    async handleSubmit(values) {
        values = this.state.fields ? this.state.fields : values.list
        let {sidenav:{listOfLinks,currentLinkPos},bulkUpdateUserAward,personalInfo:{order_data},updateCurrentLinkPos,history,showGenerateResumeModal,hideGenerateResumeModal,reGeneratePDF} = this.props
        currentLinkPos++
        this.setState({submit:true})
        await bulkUpdateUserAward(values);
        if(currentLinkPos === listOfLinks.length){
            currentLinkPos = 0
            if(order_data && order_data.id){
                showGenerateResumeModal()
                reGeneratePDF(order_data.id)
                setTimeout(function() {
                    window.location.href = `${siteDomain}/dashboard`
                    hideGenerateResumeModal()
                }, 5000);
            }
            else{
                history.push(`/resume-builder/buy`) 
            }
        }
        else{
            updateCurrentLinkPos({currentLinkPos})
            history.push(`/resume-builder/edit/?type=${listOfLinks[currentLinkPos]}`)    
        }
    }

    deleteAward(index, fields, event) {
        event.stopPropagation();
        const award = fields.get(index);
        fields.remove(index);
        if (award && award.id) {
            this.props.removeAward(award.id)
        }
    }


    editHeadingClick(){
        this.setState({editHeading:true})
    }

    componentDidUpdate(prevProps) {
        if (this.props.personalInfo.entity_preference_data !== prevProps.personalInfo.entity_preference_data) {
            this.setState({heading : this.props.personalInfo.entity_preference_data[6].entity_text})
        }
    }

    render () {
        const {handleSubmit,submitting,history,personalInfo:{order_data,entity_preference_data},changeOrderingUp,changeOrderingDown,headingChange,updateAlertModalStatus} = this.props;
        const length = parseInt(this.props.sidenav.listOfLinks.length)
        const pos = parseInt(this.props.sidenav.currentLinkPos)
        const {editHeading,heading} =this.state;
        
        return(
            <div className="buildResume">
                <PreviewModal {...this.props}/>
                <form onSubmit={handleSubmit((values)=>this.handleSubmit(values))}>
                    <FieldArray name="list" 
                                handleSubmit={handleSubmit}
                                handleAddition={this.props.handleAddition}
                                deleteAward={this.deleteAward}
                                changeOrderingUp={changeOrderingUp}
                                changeOrderingDown={changeOrderingDown}
                                component={renderAwards}
                                headingChange={headingChange}
                                editHeading={editHeading}
                                editHeadingClick={this.editHeadingClick}
                                entity_preference_data={entity_preference_data}
                                context={this}
                                heading ={heading}/>
                    <ul className="form">
                        <li className="form__group">
                            <BottomCTC  disabled={submitting} context={this} history={history} updateAlertModalStatus={updateAlertModalStatus}
                                length={length} pos={pos+1} updateInfoBeforeLoss={this.updateInfoBeforeLoss} 
                                order_data={order_data}/>
                        </li>
                    </ul>
                </form>
            </div>

        )
    }
}

export const AwardForm = reduxForm({
    form: 'award',
    validate,
    enableReinitialize: true,
    onSubmitFail: (errors) => scrollOnErrors(errors,'award',-100),
})(Award);


const mapStateToProps = (state) => {
    return {
        initialValues: state.award,
        award: state.award,
        info123: state
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        "fetchUserAward": () => {
            return dispatch(actions.fetchUserAward())
        },
        "removeAward": (awardId) => {
            return dispatch(actions.deleteAward(awardId))
        },
        "bulkUpdateUserAward": (listItems) => {
            listItems = (listItems || []).map((userAward, index) => {
                const {date} = userAward;
                if (!userAward['id']) delete userAward['id'];
                userAward = {
                    ...userAward,
                    ...{
                        date: parseInt(date)
                    }
                };
                return userAward;
            });
            return new Promise((resolve, reject) => {
                return dispatch(actions.bulkUpdateUserAward({list: listItems,resolve,reject}))
            })
        }
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(AwardForm);