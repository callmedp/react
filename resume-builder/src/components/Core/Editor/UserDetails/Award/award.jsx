import React, {Component} from 'react';
import './award.scss'


export default class Award extends Component {
    render() {
        return (
            <div>
	        </div>
        )
    }
}