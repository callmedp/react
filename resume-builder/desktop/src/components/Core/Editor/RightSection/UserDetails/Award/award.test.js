import React from 'react';
import { shallow } from 'enzyme';


import Award from './award.jsx';

describe('Award', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Award debug />);
        expect(component).toMatchSnapshot();
    });

    it('should render correctly with no props', () => {
        const component = shallow(<Award />);

        expect(component).toMatchSnapshot();
    });
});