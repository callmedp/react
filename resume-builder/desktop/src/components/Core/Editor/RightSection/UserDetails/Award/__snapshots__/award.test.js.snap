// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Award should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Award should render correctly with no props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
